community/cypher/frontend-3.4/src/main/scala/org/neo4j/cypher/internal/frontend/v3_4/PlanningAttributes.scala

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) 2002-2018 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.neo4j.cypher.internal.frontend.v3_4
+
+import org.neo4j.cypher.internal.util.v3_4.attribution.Attribute
+
+object PlanningAttributes {
+  class TransactionLayerAttribute extends Attribute[Int]
```

Comment:
As mentioned in another comment, I suggest we ditch the `...Attribute` suffix from this class. I think it might hurt more than help. Maybe attributes should have plural types to indicate that the instance has the attribute of every node in the tree? E.g. `TransactionLayers` in this case.